15_perms_lab/permslab.txt

```diff
@@ -1,15 +1,17 @@
 1. Using the file structure in the current directory, write the commands that perform the following:
 
   - change permission of file src/main.c to r--r--r-- using chmod letter notation
-
+chmod a=r src/main.c
   - change permission of file include/player.h to r-xrw--r- using chmod octal (numbers) notation
-
+chmod 564 src/player.c
   - change permission of all files in Doc folder (without changing the permissions of Doc folder itself) to rw-------
-
+chmod 777 Doc/*.*
   - change permission of "lib" folder and all files and subdirectories inside it to rwxrwxrwx
-
+chmod 777 l*/*.* or chmod -R 777 lib
 2. create a new directory named "test" and inside an empty file named "secret". Change permissions of the file "secret" to ---------. Can you delete the file secret? can you delete the folder test? why?
-
+I can delete the empty file: rm test/secret
+Cannot delete test, since it is a folder (Need to use rm -r)
 3. Read the output of "help umask". Explain in your own words what umask does, and how would you use it to set new files permissions to ------r--
-
+Gives you the option to change the defaults riths upon creation of files and folders
+There is a need to change the umask to 003, in order to set the files permissions to ------r--
```

Comment:
umask sets a negative bit mask, so we'll want the opposite. An enabled bit in the umask is actually a `-`, i.e. no permission on the result. So the correct umask here is:
`umask 662`
Turning off rw for owner and group, and write for other. What's left is just the "r" permission for others